### PR TITLE
feat: allow admins to set scope for requests

### PR DESCRIPTION
**Pull request by vanyae-cqc**
Also relaxes dependency version constraints.

### CODE REVIEW

File: qnexus/client/circuits.py [hunk 1]
```diff
@@ -72,11 +76,12 @@ def get_all(  # pylint: disable=too-many-positional-arguments
         sort=SortFilter.convert_sort_filters(sort_filters),
         page_number=page_number,
         page_size=page_size,
+        scope=scope,
     ).model_dump(by_alias=True, exclude_unset=True, exclude_none=True)
 
     return NexusIterator(
         resource_type="Circuit",
-        nexus_url="/api/circuits/v1beta",
+        nexus_url="/api/circuits/v1beta?scope=global_admin",
```
Comment (quantinuum-richard-morrison) [inline qnexus/client/circuits.py:84]: Is this intentional?  `?scope=global_admin` 
Or just leftover from development experiments.

File: qnexus/client/projects.py [hunk 2]
```diff
@@ -154,7 +161,9 @@ def get_or_create(
 
 def _fetch_by_id(project_id: UUID | str) -> ProjectRef:
     """Utility method for fetching directly by a unique identifier."""
-    res = get_nexus_client().get(f"/api/projects/v1beta/{project_id}")
+    res = get_nexus_client().get(
+        f"/api/projects/v1beta/{project_id}?scope=global_admin"
+    )
```
Comment (quantinuum-richard-morrison) [inline qnexus/client/projects.py:166]: Same comment as before re `?scope=global_admin`

File: qnexus/client/circuits.py (hunk 1 above)
Comment (swhiteCQC) [inline qnexus/client/circuits.py:84]: What's the default behaviour if a user sets the scope to global_admin without having that level of permission?

File: qnexus/client/projects.py (hunk 2 above)
Comment (vanyae-cqc) [reply qnexus/client/projects.py:166]: Ah I'm sorry, indeed thats left over. Will fix the sloppy PR!

**Comment by vanyae-cqc** [comment]
Thanks @quantinuum-richard-morrison - added a simple integration test for the filter

**Pull request merged**